File: 2391-minimum-amount-of-time-to-collect-garbage/2391-minimum-amount-of-time-to-collect-garbage.cpp
class Solution {
public:

    int  collection_time(vector<string>& garbage, vector<int>& travel,char ch) 
    {
        int count=0;
        int time=0;
        int last=0;
        for(int i=0;i<garbage.size();i++)
        {
            for(int k=0;k<garbage[i].size();k++)
            {
                if(garbage[i][k] == ch)
                {
                    last=i;
                    count++;
                }
            }

        }
        for(int i=0;i<last;i++)
        {
            time+=travel[i];
        }
        return time+count;
    }
    int garbageCollection(vector<string>& garbage, vector<int>& travel) {
     int time=0;
      time +=  collection_time(garbage,travel,'G');
      time +=  collection_time(garbage,travel,'M');
      time +=  collection_time(garbage,travel,'P');

    return time;
        
    }
};

File: 2391-minimum-amount-of-time-to-collect-garbage/2391-minimum-amount-of-time-to-collect-garbage.java
class Solution {
  
    int time(String[] garbage, int[] travel,char ch)
    {
        int last=0;
        int count=0;
        for(int i=0;i<garbage.length;i++)
        {
            for(int j=0;j<garbage[i].length();j++)
            {
                if(garbage[i].charAt(j)==ch)
                {
                    last=i;
                    count++;
                }
            }
        }
        int time=0;
        for(int i=0;i<last;i++)
        {
            time += travel[i];
        }
        return time+count;

    }
    public int garbageCollection(String[] garbage, int[] travel) {
        int total_time=0;
         total_time +=time(garbage,travel,'G');
         total_time +=time(garbage,travel,'P');
         total_time +=time(garbage,travel,'M');
         return total_time;
        
    }
}